### PR TITLE
[release-v1.0] Copy Broker and KafkaSink improvements

### PR DESCRIPTION
**Pull request by pierDipi**
In this PR, we're taking these upstream fixes (some might already be in this branch):
https://github.com/knative-sandbox/eventing-kafka-broker/commit/87fe43936a0f5023f5645f67b9c00a01d0b61155
https://github.com/knative-sandbox/eventing-kafka-broker/commit/f32703881dc58aaa5d6757c6be4b9f31cd5f5f48
https://github.com/knative-sandbox/eventing-kafka-broker/commit/d5f07bb07591c1d26fbe54f4882466296155e7ce
https://github.com/knative-sandbox/eventing-kafka-broker/commit/67f09936d5ced6159d2a8407e29286476a7366c4
https://github.com/knative-sandbox/eventing-kafka-broker/commit/da9c0fbf04dfd3f8f5eb0efd4d48f7f577176bc4
https://github.com/knative-sandbox/eventing-kafka-broker/commit/35d74b0921661b612d0da1f178277b34612b9eba
https://github.com/knative-sandbox/eventing-kafka-broker/commit/dd934169b3579811ee9b62f79465dd4bb9d8cf4f
https://github.com/knative-sandbox/eventing-kafka-broker/commit/f04a70f0001f89f5d58231f234e40c309329500a
https://github.com/knative-sandbox/eventing-kafka-broker/commit/0e5912f8e44d100e760c032fefdbece294826d79
https://github.com/knative-sandbox/eventing-kafka-broker/commit/c56aae3967b26c1796319b13e55a4a2f54649f5a
https://github.com/knative-sandbox/eventing-kafka-broker/commit/b7b3f70f4db012d65a83d1c3d9df10dd0e1a4739
https://github.com/knative-sandbox/eventing-kafka-broker/commit/6a4ccc00c52fd100a1c75d68206ddc3d7efdad43
https://github.com/knative-sandbox/eventing-kafka-broker/commit/6899909620d1db2ab2477f7e4de8a6cd909f768a
https://github.com/knative-sandbox/eventing-kafka-broker/commit/7605d595c4269e748d95e4995b781e9bb978631d

### CODE REVIEW

File: openshift/release/generate-release.sh [hunk 1]
```diff
@@ -30,14 +30,8 @@ resolve_resources control-plane/config/200-webhook cp_broker.yaml $image_prefix
 cat cp_broker.yaml >> $broker_cp_output_file
 rm cp_broker.yaml
 
-# the Broker Data Plane folders
-# The generic DP root folder
-resolve_resources data-plane/config $broker_dp_output_file $image_prefix $tag
-
-# The DP folder for Broker:
-resolve_resources data-plane/config/broker dp_broker.yaml $image_prefix $tag
-cat dp_broker.yaml >> $broker_dp_output_file
```
Comment (matzew) [inline openshift/release/generate-release.sh:39]: are you sure we really want all this? :see_no_evil:

File: vendor/knative.dev/reconciler-test/pkg/images/ko.go [hunk 2]
```diff
@@ -16,7 +16,20 @@ limitations under the License.
...
+import (
+	"fmt"
+	"os"
+)
+
 // Use ko to publish the image.
 func KoPublish(path string) (string, error) {
-	return "", nil
+	platform := os.Getenv("PLATFORM")
+	if len(platform) > 0 {
```
Comment (matzew) [inline vendor/knative.dev/reconciler-test/pkg/images/ko.go:27]: no, we did the patch for a reason

**Comment by matzew** [comment]
I have some concerns as this touches:
* `openshift` and `vendor` folder.  `./hack/update-deps.sh` does not really work here ...

**Comment by matzew** [comment]
Let's identify what we really want, and port them over - one-by-one

**Comment by pierDipi** [comment]
We need proper patches in the `openshift/patches/` directory, so far updates to vendor folder were done directly to the file

**Comment by matzew** [comment]
/hold
do not merge

**Comment by matzew** [comment]
So.... for 1.0 let's please collect all PRs we _really_ need (not nice to have).
and do them here.

For 1.1+ we can see that we have a better approach (we have more time)

**Comment by pierDipi** [comment]
We need all of them, these are bug fixes.

**Comment by pierDipi** [comment]
> > We need proper patches in the `openshift/patches/` directory, so far updates to vendor folder were done directly to the file
> 
> yes, I know - and that's b/c we started this a WIP/TP component. I'd really rather get this fixed on `release-next` and forward (and perhaps 1.2).

That's what I'm gonna do next (we don't have issues tracking this), but I'm confused on why you want to fix only main, next and 1.2

**Comment by matzew** [comment]
> That's what I'm gonna do next (we don't have issues tracking this), but I'm confused on why you want to fix only main, next and 1.2

For 1.2 - I am still in the middle of setting that up. Is it possible I do it there, and than we apply it down? (and up)?

**Comment by matzew** [comment]
@pierDipi looks like this is needing a rebase now

**Comment by matzew** [comment]
@pierDipi can we get some better description here, and I think than we are good to go

**Comment by matzew** [comment]
the actual `.patch` file has still old references.... 

but looking like actual files, they look correct - example:
https://github.com/pierDipi/eventing-kafka-broker/blob/copy-improvements-v1.0/vendor/knative.dev/eventing/test/rekt/resources/eventlibrary/eventlibrary.yaml#L26

**Comment by pierDipi** [comment]
> the actual `.patch` file has still old references....
> 

I'm not sure I get what you mean :thinking: 

> but looking like actual files, they look correct - example: https://github.com/pierDipi/eventing-kafka-broker/blob/copy-improvements-v1.0/vendor/knative.dev/eventing/test/rekt/resources/eventlibrary/eventlibrary.yaml#L26

**Comment by pierDipi** [comment]
> @pierDipi can we get some better description here, and I think than we are good to go

I added some upstream commits in the PR body.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**